doc/adr/0003-self-hosted.md

```diff
@@ -0,0 +1,86 @@
+# 3. Self Hosted
+
+Date: 2021-04-27
+
+## Status
+
+Proposed
+
+## Context
+
+We want to enable enterprise customers to self host Athens knowledge graph on premise.  
+Uses should be able to collaborate on same graph, edit block & pages as well as see presence of other collaborators.
+
+## Decision
+
+We need a backend server that at least talks Datalog.
+
+For 1st backend we've chosen Datahike, because it's closeness to Datascript that we already use.  
+Another possible backend could be Crux or Datomic.
+
+### Communication Layer
+
+Because we're running in a very controlled environment (as in we don't have to support Samsung Internet & FB embedded browser)  
+we can assume availability of modern Web APIs, so for client server communication we can rely purely on WebSockets.
+
+### Concurrency
+
+To avoid problems of concurrent updates, we'd like to have single writer approach.  
+This means that updates to central graph can be done synchronously.
+
+### Connecting clients
+
+When client connects we have to get it in sync with backend.  
+
+We have 2 options here:
+
+1. Stream all the updates that happened since last tx-id client has. (easier to do because we'll need tx ingest on client for normal connected work)
+2. Provide a snapshot of DB state. (more effective on busy DB)
+
+We also have to authorize user.  
+For now we can assume trust based system, so if you say you're "Plato" we believe you w/o verification.  
```

Comment:
the problem of deciding "is the user actually Plato as they claim" is authentication. "is the user allowed to access this resource" is authorization.

(see e.g. https://docs.microsoft.com/en-us/azure/active-directory/develop/authentication-vs-authorization)